test/phoenix_client_test.exs

```diff
@@ -297,6 +297,18 @@ defmodule PhoenixClientTest do
     refute Socket.connected?(socket)
   end
 
+  test "socket reports back on connect if requested", context do
+    endpoint = context[:endpoint]
+    config =
+      @socket_config
+      |> Keyword.put(:report_back_pid, self())
+      |> Keyword.put(:reconnect_interval, 100)
+
+    {:ok, socket} = Socket.start_link(config)
+    wait_for_socket(socket)
+    assert_received {:phoenix_client, :connected}
+  end
```

Comment:
I tried many ways of testing the `:disconnected` message, but there's some complicated OTP stuff going on here and it's interacting with the other tests even though `async` is `false`. 😭